src/components/ChatWindow/ChatWindow.tsx

```diff
@@ -45,13 +58,42 @@ const ChatWindow = ({ onChatActivation }: Props) => {
     }
   }
 
+  const handleClearChat = () => {
+    // prevent function from running when there is no chat
+    if (!chatHistory.find((chat) => chat.sender === "user")) {
+      close()
+      return
+    }
+
+    setChatHistory(DEFAULT_MSG)
+    close()
+  }
+
   return (
     <Paper shadow="sm" withBorder className={styles.chatWindow}>
+      <Modal
+        opened={openedModal}
+        onClose={close}
+        className={styles.modalContainer}
+        withinPortal={false}
+      >
```

Comment:
Can we make a reusable Modal component using Mantine's modal like this? Just pass the props in and use it instead of building the modal directly in the ChatWindow

Reply:
Solved it in `commit 6749744ca271fbd8675855226468ac18d6de4d74`. 

Now ChatWindow contains:
+ Re-usable ConfirmationModal
+ `useDisclosure` hook that is used for Modal